prospero-cli/src/main/java/org/wildfly/prospero/cli/RepositoryDefinition.java

```diff
@@ -17,32 +17,45 @@
 
 package org.wildfly.prospero.cli;
 
+import org.jboss.logging.Logger;
 import org.wildfly.channel.Repository;
 
+import java.io.File;
 import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
 
 public class RepositoryDefinition {
 
+    private static final Logger logger = Logger.getLogger(RepositoryDefinition.class.getName());
+
     public static List<Repository> from(List<String> repos) throws ArgumentParsingException {
         ArrayList<Repository> repositories = new ArrayList<>(repos.size());
         for (int i = 0; i < repos.size(); i++) {
-            final String text = repos.get(i);
-            if(text.contains("::")) {
-                final String[] parts = text.split("::");
+            String repoName = repos.get(i);
```

Comment:
nitpick - the `repoName` variable name is a bit misleading - the string can either contain repoName::repoLocation or just repoLocation